pallets/pools/src/lib.rs

```diff
@@ -615,6 +616,23 @@ pub mod pallet {
 			let now = Self::now();
 			let tranches = Tranches::from_input::<T::TrancheToken>(pool_id, tranches, now)?;
 
+			let checked_metadata: Option<BoundedVec<u8, T::MaxSizeMetadata>> = match metadata {
+				Some(metadata_value) => {
+					let checked: BoundedVec<u8, T::MaxSizeMetadata> = metadata_value
+						.clone()
+						.try_into()
+						.map_err(|_| Error::<T>::BadMetadata)?;
+
+					Self::deposit_event(Event::MetadataSet {
```

Comment:
I am thinking whether this is a good idea. Nothing is actually _set_ here, I assume we just deposit this event at this point because here we know that `metadata` is `Some`.

Why do we want to deposit this event when creating a pool? At this point, I'd rather have the `Created` event containing a new field `metadata: Option<Vec<u8>>`

Reply:
My reasoning was that a client that wants to know whenever the metadata changes, should only have to listen to the `MetadataSet` event. But I realized that this is actually inconsistent with the other `create` behaviour, we should then also submit `Updated`, `Rebalanced`, `MaxReserveSet` events. So I think it makes sense to remove it then.